commercelayer/resource_adyen_gateway.go

```diff
@@ -0,0 +1,204 @@
+			"(like secret and publishable keys, etc. — contact Adyen's support if you are not sure about the requested data).",
+		ReadContext:   resourceAdyenGatewayReadFunc,
+		CreateContext: resourceAdyenGatewayCreateFunc,
+		UpdateContext: resourceAdyenGatewayUpdateFunc,
+		DeleteContext: resourceAdyenGatewayDeleteFunc,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "The adyen payment unique identifier",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"type": {
+				Description: "The resource type",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"attributes": {
+				Description: "Resource attributes",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				MinItems:    1,
+				Required:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Description: "The payment gateway's internal name.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"merchant_account": {
+							Description: "The gateway merchant account.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"api_key": {
+							Description: "The gateway API key.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"public_key": {
+							Description: "The public key linked to your API credential.",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"live_url_prefix": {
+							Description: "The prefix of the endpoint used for live transactions.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"reference": {
+							Description: "A string that you can use to add any external identifier to the resource. This " +
+								"can be useful for integrating the resource to an external system, like an ERP, a " +
+								"marketing tool, a CRM, or whatever.",
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"reference_origin": {
+							Description: "Any identifier of the third party system that defines the reference code",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"metadata": {
+							Description: "Set of key-value pairs that you can attach to the resource. This can be useful " +
+								"for storing additional information about the resource in a structured format",
+							Type: schema.TypeMap,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+							Optional: true,
+						},
+						"relationships": {
+							Description: "Resource relationships",
+							Type:        schema.TypeList,
+							MaxItems:    1,
+							MinItems:    1,
+							Optional:    true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"adyen_payment_id": {
```

Comment:
I don't actually think we want this in the provider, as it is stuff that actually should be created only when we have a running platform (like products etc). Can you remove it?